src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -873,6 +873,42 @@ extends scala.collection.concurrent.Map[K, V]
     insertifhc(k, hc, v, INode.KEY_ABSENT)
   }
 
+  // TODO once this is added to concurrent.Map, move the comment there
+  /** If the specified key is not already in the map, computes its value using
+   *  the given mapping function `f` and enters it into the map.
+   *
+   *  Since concurrent maps cannot contain `null` for keys or values,
+   *  a `NullPointerException` is thrown if the given mapping function `f`
+   *  returns `null`.
+   *
+   *  If the specified mapping function throws an exception,
+   *  that exception is rethrown.
+   *  
+   *  Note: this method may invoke the specified function `f` more than once,
+   *  even if the key is in the end not added to the map.
+   *
+   *  @param k      the key to modify
+   *  @param f      the mapping function that computes the value from the key
+   *  @return       the newly added value
+   */
+  def computeIfAbsent(k: K, f: K => V): V = {
```

Comment:
I've chosen this name and signature mainly to be consistent with the `computeIfAbsent` in `java.util.concurrent`, the way we've done it for pre-JDK8 for other `ConcurrentMap` methods. My guess is that eventually we want to do this, to ease the conversions between Java and Scala concurent maps.

I see your concern, however. I could just keep `getOrElseUpdate` in this PR.